cohere/responses/custom_model.py

```diff
@@ -73,6 +74,23 @@ def from_dict(cls, data: Dict[str, Any]) -> "CustomModel":
             model_id=data["model"]["route"] if "model" in data else None,
         )
 
+@dataclass
```

Comment:
might be easier to have `CustomModel` as a dataclass as well, would omit having to write out the init method.